src/videojs-http-streaming.js

```diff
@@ -427,6 +427,64 @@ const expandDataUri = (dataUri) => {
   return dataUri;
 };
 
+/**
+ * Adds a request hook to an xhr object
+ *
+ * @param {Object} xhr object to add the onRequest hook to
+ * @param {function} callback hook function for an xhr request
+ */
+const addOnRequestHook = (xhr, callback) => {
+  if (!xhr.onRequest) {
+    xhr.onRequest = new Set();
+  }
+  xhr.onRequest.add(callback);
+};
+
+/**
+ * Adds a response hook to an xhr object
+ *
+ * @param {Object} xhr object to add the onResponse hook to
+ * @param {function} callback hook function for an xhr response
+ */
+const addOnResponseHook = (xhr, callback) => {
+  if (!xhr.onResponse) {
```

Comment:
these probably should be pseudo-private properties. Maybe something like `xhr._responseCallbacks`